frontend/src/components/FormElements/TimeField.jsx

```diff
@@ -0,0 +1,175 @@
+import { useState, useRef } from "react";
+import TextField from "@mui/material/TextField";
+import { Controller } from "react-hook-form";
+import { Grid, Box, Typography, FormHelperText } from "@mui/material";
+
+const TimeField = ({ label, name, control, rules }) => {
+  const inputRefs = useRef([null, null, null]);
+
+  return (
+    <Controller
+      name={name}
+      control={control}
+      rules={rules}
+      render={({ field: { onChange, value }, fieldState: { error } }) => {
+        const [time, setTime] = useState({
+          minutes: "",
+          seconds: "",
+          milliseconds: "",
+        });
+
+        const handleChange = (field, newValue, currentIndex) => {
+          const trimmedValue = newValue.slice(0, 2);
+          setTime((prevTime) => {
+            const updatedTime = { ...prevTime, [field]: trimmedValue };
+            const formattedTime = `${
+              updatedTime.minutes.padStart(2, "0") || "00"
+            }:${updatedTime.seconds.padStart(2, "0") || "00"}.${
+              updatedTime.milliseconds.padStart(2, "0") || "00"
+            }`;
+            if (formattedTime === "00:00.00") {
+              onChange(null);
+            } else {
+              onChange(formattedTime);
+            }
+            if (
+              trimmedValue.length === 2 &&
+              currentIndex < inputRefs.current.length - 1
+            ) {
+              inputRefs.current[currentIndex + 1]?.focus();
+            }
+            return updatedTime;
+          });
+        };
```

Comment:
I think we can use only `newValue` and avoid having _trimmedValue_  since the `maxLength` for each field is 2 and also in line 36, if we use `newValue.length === 2` will also allow the user to enter only 2 digits before focus in the next field.